data/src/history.rs

```diff
@@ -75,7 +75,8 @@ impl From<String> for Kind {
 
 impl From<&str> for Kind {
     fn from(target: &str) -> Self {
-        if proto::is_channel(target) {
+        // XXX not sure how to fix this in a way that will preserve on-disk history
+        if proto::is_channel(target, proto::DEFAULT_CHANNEL_PREFIXES) {
```

Comment:
We only construct via this when receiving read marker updates from the server, which will always be a valid channel or query.

I we remove this completely once we get around to refactoring as I mention in https://github.com/squidowl/halloy/pull/623/files#r1809548087 and we add some new `Channel` and `Target` types to use across the codebase